Shared/Samples/Show line of sight between geoelements/ShowLineOfSightBetweenGeoelementsView.swift

```diff
@@ -0,0 +1,344 @@
+                Button("Settings") {
+                    isPresented = true
+                }
+                .sheet(isPresented: $isPresented) {
+                    settingsSheet
+                }
+            }
+        }
+    }
+    
+    /// The settings configuration sheet for adjusting the observer's height.
+    private var settingsSheet: some View {
+        NavigationStack {
+            Form {
+                let heightRange = 20.0...70.0
+                let numberFormat = FloatingPointFormatStyle<Double>.number.precision(.fractionLength(0))
+                LabeledContent(
+                    "Observer Height",
+                    value: model.height,
+                    format: numberFormat
+                )
+                Slider(
+                    value: $model.height,
+                    in: heightRange,
+                    step: 1
+                ) {
+                    Text("Observer Height")
+                } minimumValueLabel: {
+                    Text(heightRange.lowerBound, format: numberFormat)
+                } maximumValueLabel: {
+                    Text(heightRange.upperBound, format: numberFormat)
+                }
+                .listRowSeparator(.hidden, edges: .top)
+            }
+            .presentationDetents([.fraction(0.25)])
+            .navigationTitle("Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        isPresented = false
+                    }
+                }
+            }
+        }
+    }
+}
+
+private extension ShowLineOfSightBetweenGeoelementsView {
+    /// View model responsible for setting up the 3D scene, graphics, animation,
+    /// and performing line of sight analysis between moving and static geoelements.
+    @MainActor
+    @Observable
+    final class Model {
+        /// A set of predefined waypoints for animating the taxi's movement.
+        private let points = [
+            Point(latitude: 40.748469, longitude: -73.984513),
+            Point(latitude: 40.747786, longitude: -73.985068),
+            Point(latitude: 40.747091, longitude: -73.983452),
+            Point(latitude: 40.747762, longitude: -73.982961)
+        ]
+        
+        /// The height of the observer in meters. Updates the observer graphic when changed.
+        var height = 20.0 {
+            didSet {
+                changeObserverHeight(height)
+            }
+        }
+        
+        /// The current frame number in the animation sequence for the taxi's movement.
+        private var frameIndex = 0
+        
+        /// The total number of animation frames to complete a segment between two waypoints.
+        private let frameMax = 120
+        
+        /// The index of the current start point in the `points` array used for animating the taxi's path.
+        private var pointIndex = 0
+        
```

Comment:
<img width="455" alt="Screenshot 2025-07-09 at 10 18 04 AM" src="https://github.com/user-attachments/assets/9d9e702e-f66a-4fdf-9ad0-edf3f110d74e" />

In the current version of SwiftUI, `ObservationTracked` is automatically [added](https://developer.apple.com/forums/thread/763664?answerId=803693022) to all variables in an Observable object. Therefore, it would be good to add `@ObservationIgnored` to variables that don't need to be published, although it is up to you (coding style aesthetics vs precision) to add it or not as the performance gain is minimum. We can discuss more if you want.

Reply:
This is interesting! I definitely want to learn more. I've added it in on my local branch and tested it. I think if it's a best practice and improves performance it's a good addition, small things add up. My only question is if the pattern is well known enough or would generate confusion.